linode/manager
# fix: [M3-6021] - Invalidate Firewall devices cache when a Linode is deleted

## Description 📝

- What a Linode that was associated with a firewall was deleted, the **Firewalls** landing and details page would not update to reflect the removed Linode.
- This PR React Query-ifys the Firewall Devices endpoint and performorms a cache invalidation on it whenever a Linode is deleted that is associated with that particular firewall
- Cleans up and modernizes some components in the process

## Preview 📷

### Before

https://user-images.githubusercontent.com/115251059/223291995-5ebd4ed0-e5ef-4ed1-80a7-a90fb37d8cf3.mov

### After

https://user-images.githubusercontent.com/115251059/223292005-a6bf3bff-58f5-43e8-9a90-5a489bf125ba.mov

## How to test 🧪

- Have at least 1 Firewall and 1 Linode
- Attach the Linode to the Firewall
- Go delete that Linode
- Go back to the Firewall landing page
- Observe that the Linode is no longer listed under that Firewall
